### DISCUSSION TITLE
Getting the X window id of a window

### DISCUSSION THREAD

**Discussion by foodornt** [wezterm/wezterm#3971] [Q&A]
TLDR; Window ID 1 = X Window ID ?

To not make this an XY problem:
I can list windows available along with their titles by issuing the command `wezterm cli list --format=json`, and that'll give me window titles along with tab titles

I want to get the window ID of the window that's holding a certain tab (with a certain title) so I can make my window manager navigate to it

any idea? even a general outline of what I should do is an acceptable answer, thanks in advance <3

**Comment by wez** [comment]
we'll never support specifically exposing X window IDs, but look instead at:
https://wezfurlong.org/wezterm/cli/cli/activate-pane.html

**Reply by foodornt** [reply]
Fair enough, will activate-pane trigger an EWMH activate event? I'm using AwesomeWM, and it doesn't seem to be receiving any events upon issuing `wezterm cli activate-tab --tab-id=X`

**Reply by wez** [reply]
did you try activate-pane rather than activate-tab? They work differently

**Reply by foodornt** [reply]
Yes, both don't send EWMH activation events

**Reply by wez** [reply]
Please file a bug report with all the detail needed to reproduce and analyze this

**Reply by foodornt** [reply]
Will do 👍
